### DISCUSSION TITLE
Is there a way to control the printer to print the Request Body and Response Body only on failure?

### DISCUSSION THREAD

**Discussion by arvind-sys** [gavv/httpexpect#138] [Q&A] [answered]
Is there a way to control the printer to print the Request Body and Response Body only on failure?
As of now the Request and response body are printed for all the Tests. Is there a way to control it in order to print the same only when the tests fail?

**Comment by gavv** [comment, marked as answer]
Hi. Unfortunately, there is no way to do it.

I guess it would not be easy to implement it, since currently we're printing request and response before constructing Response objects, on which the user performs assertions.

To achieve the behavior you need, we should maintain some context with both request and response and pass it to every object that may perform assertions (e.g. Value, Object, Array, etc.). More specifically, we should add it to `chain` object and forward to child chains. Then `chain` can use this context to print desired info on failure.

Here is where request and response are printed currently: https://github.com/gavv/httpexpect/blob/0387206792097a6dc6fd830ac268a6f1953194f1/request.go#L893-L917

**Comment by gavv** [comment]
https://github.com/gavv/httpexpect/issues/159
